### PR TITLE
Splitter: support cancellable resize events

### PR DESCRIPTION
**Pull request by AlexanderMoiseev**

### CODE REVIEW

File: e2e/testcafe-devextreme/tests/navigation/splitter/events.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import url from '../../../helpers/getPageUrl';
+import { createWidget } from '../../../helpers/createWidget';
+import Splitter from 'devextreme-testcafe-models/splitter';
+
+fixture.disablePageReloads`Splitter_events`
+  .page(url(__dirname, '../../container.html'));
+
+  test('Panes should not be able to resize when onResizeStart event canceled', async (t) => {
```
Comment (EugeniyKiyashko) [inline e2e/testcafe-devextreme/tests/navigation/splitter/events.ts:8]: Why we can't to write this test using qunit?

Comment (AlexanderMoiseev) [reply e2e/testcafe-devextreme/tests/navigation/splitter/events.ts:8]: have not managed to do it with pointerMock's drag dragStart etc, but probablty pointer.down().move will work, i will try again

Comment (AlexanderMoiseev) [reply e2e/testcafe-devextreme/tests/navigation/splitter/events.ts:8]: i tried and we decided to go with testcafe test

File: packages/devextreme/js/__internal/ui/splitter/splitter.ts [hunk 2]
```diff
@@ -516,28 +533,28 @@ class Splitter extends (CollectionWidget as any) {
...
 
         const $resizeHandle = $(element);
 
+        this._getAction(RESIZE_EVENT.onResizeEnd)({
+          event,
+          handleElement: getPublicElement($resizeHandle),
+        });
+
+        // @ts-expect-error typify event members
+        if (event.cancel) {
```
Comment (EugeniyKiyashko) [inline packages/devextreme/js/__internal/ui/splitter/splitter.ts]: Why we check native event arguments? We should check cancel flag in action arguments. In this case // @ts-expect-error doesn't require

File: packages/devextreme/js/__internal/ui/splitter/splitter.ts [hunk 3]
```diff
@@ -476,8 +476,20 @@ class Splitter extends (CollectionWidget as any) {
...
-        this._currentLayout = this._layout;
         const $resizeHandle = $(element);
+
+        this._getAction(RESIZE_EVENT.onResizeStart)({
+          event,
+          handleElement: getPublicElement($resizeHandle),
+        });
+
+        // @ts-expect-error typify event members
+        if (event.cancel) {
```
Comment (Zedwag) [inline packages/devextreme/js/__internal/ui/splitter/splitter.ts]: `cancel` should be propetry of main arg, not it's `event` field. For example [link](https://js.devexpress.com/jQuery/Documentation/ApiReference/UI_Components/dxDraggable/Types/DragStartEvent/#cancel)

**Review by EugeniyKiyashko — APPROVED** [review]

**Pull request merged**